pyffs/ffs.py

```diff
@@ -592,3 +595,262 @@ def iffsn(Phi_FS, T, T_c, N_FS, axes=None):
+
+    Its Fourier Series (FS) coefficients :math:`\phi_{k_x, k_y}^{FS}` can be analytically evaluated
+    using the shift-modulation theorem:
+
+    .. math::
+
+       \phi_{k_x, k_y}^{FS} =
+       \begin{cases}
+           \exp\left( -j \frac{2 \pi}{T_x} k_x T_{c,x} \right) \exp\left( -j \frac{2 \pi}{T_y} k_y
+           T_{c,y} \right) & -N_x \le k_x \le N_x, -N_y \le k_y \le N_y,  \\
+           0 & \text{otherwise}.
+       \end{cases}
+
+    Being bandlimited, we can use :py:func:`~pyffs.ffs2` to numerically evaluate :math:`\{\phi_{k_x,
+    k_y}^{FS}, k_x = -N_x, \ldots, N_x, k_y = -N_y, \ldots, N_y\}`:
+
+    .. testsetup::
+
+       import math
+
+       import numpy as np
+
+       from pyffs import ffsn_sample, ffsn
+
+       def dirichlet(x, T, T_c, N_FS):
+           y = x - T_c
+
+           n, d = np.zeros((2, len(x)))
+           nan_mask = np.isclose(np.fmod(y, np.pi), 0)
+           n[~nan_mask] = np.sin(N_FS * np.pi * y[~nan_mask] / T)
+           d[~nan_mask] = np.sin(np.pi * y[~nan_mask] / T)
+           n[nan_mask] = N_FS * np.cos(N_FS * np.pi * y[nan_mask] / T)
+           d[nan_mask] = np.cos(np.pi * y[nan_mask] / T)
+
+           return n / d
+
+       def dirichlet_2D(sample_points, T, T_c, N_FS):
+
+           # compute along x and y, then combine
+           x_vals = dirichlet(x=sample_points[0][:, 0], T=T[0], T_c=T_c[0], N_FS=N_FS[0])
+           y_vals = dirichlet(x=sample_points[1][0, :], T=T[1], T_c=T_c[1], N_FS=N_FS[1])
+           return np.outer(x_vals, y_vals)
+
+       def dirichlet_fs(N_FS, T, T_c):
+           N = (N_FS - 1) // 2
+           return np.exp(-1j * (2 * np.pi / T) * T_c * np.r_[-N : N + 1])
+
+    .. doctest::
+
+       >>> T = [1, 1]
+       >>> T_c = [0, 0]
+       >>> N_FS = [3, 3]
+       >>> N_s = [4, 3]
+
+       # Sample the kernel and do the transform.
+       >>> sample_points, _ = ffsn_sample(T=T, N_FS=N_FS, T_c=T_c, N_s=N_s)
+       >>> diric_samples = dirichlet_2D(sample_points, T, T_c, N_FS)
+       >>> diric_FS = ffsn(Phi=diric_samples, T=T, N_FS=N_FS, T_c=T_c)
+
+       # Compare with theoretical result.
+       >>> diric_FS_exact = np.outer(
+       ... dirichlet_fs(N_FS[0], T[0], T_c[0]), dirichlet_fs(N_FS[1], T[1], T_c[1])
+       ... )
+       >>> np.allclose(diric_FS[: N_FS[0], : N_FS[1]], diric_FS_exact)
+       True
+
+    Notes
+    -----
+    Theory: :ref:`FFS_def`.
+
+    See Also
+    --------
+    :py:func:`~pyffs.util.ffsn_sample`, :py:func:`~pyffs.ffs.iffsn`
+    """
+
+    if axes is None:
+        D = len(Phi.shape)
+        axes = tuple(range(D))
+    else:
+        D = len(list(set(axes)))
+        assert D == len(axes), "[axes] must contain unique values."
+
+    # check same length
+    assert len(T) == D, "Length of [T] must match dimension of [Phi]."
+    assert len(T_c) == D, "Length of [T_c] must match dimension of [Phi]."
+    assert len(N_FS) == D, "Length of [T] must match dimension of [Phi]."
+
+    # check valid values
+    N_s = []
```

Comment:
`N_s = np.array(Phi.shape)[axes]` should work too.
This keeps the loop to error-checking only.

Reply:
done inside `pyffs.util._verify_ffsn_input`